app/src/main/kotlin/com/wire/android/ui/home/userprofile/UserProfileScreen.kt

```diff
@@ -1,16 +1,343 @@
 package com.wire.android.ui.home.userprofile
 
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBarDefaults
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.tooling.preview.Preview
-import com.wire.android.ui.common.UnderConstructionScreen
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import com.wire.android.R
+import com.wire.android.navigation.NavigationManager
+import com.wire.android.ui.common.Icon
+import com.wire.android.ui.common.UserProfileAvatar
+import com.wire.android.ui.common.UserStatus
+import com.wire.android.ui.common.UserStatusDot
+import com.wire.android.ui.common.button.WireButtonState
+import com.wire.android.ui.common.button.WireSecondaryButton
+import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.common.selectableBackground
+import com.wire.android.ui.common.textfield.WirePrimaryButton
+import com.wire.android.ui.theme.wireTypography
 
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
-fun UserProfileScreen() {
-    UnderConstructionScreen(screenName = "UserProfileScreen")
+fun UserProfileScreen(viewModel: UserProfileViewModel) {
+    val uiState by viewModel.state.collectAsState()
+
+    with(uiState) {
+        renderUserProfileScreen(state = this, viewModel = viewModel)
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun renderUserProfileScreen(state: UserProfileState, viewModel: UserProfileViewModel) {
```

Comment:
I'm not sure about that, because here the state is extracted from the viewmodel but still the viewmodel is also passed - what I prefer is to separate it and don't pass the viewmodel to the inner composables, instead pass the state and all "onclick" functions as parameters - this way we don't need a viewmodel to create a preview and test it in the future, but it's my personal opinion

Reply:
Or even keep all the elements that depend on the VM within the same Composable function scope